### PR TITLE
Merging master changes into csharp and adding initial C# implementation

### PR DESCRIPTION
**Pull request by fabiocav**

### CODE REVIEW

File: src/WebJobs.Script.WebHost/Controllers/HomeController.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Net;
+using System.Net.Http;
+using System.Web.Http;
+
+namespace WebJobs.Script.WebHost.Controllers
+{
+    public class HomeController : ApiController
```
Comment (MikeStall) [inline src/WebJobs.Script.WebHost/Controllers/HomeController.cs]: What is this controller for?

File: src/WebJobs.Script/Binding/BlobBinding.cs [hunk 2]
```diff
@@ -38,17 +38,25 @@ public override bool HasBindingParameters
 
             boundBlobPath = Resolve(boundBlobPath);
 
-            // TODO: Need to handle Stream conversions properly
-            Stream valueStream = context.Value as Stream;
-
-            Stream blobStream = context.Binder.Bind<Stream>(new BlobAttribute(boundBlobPath, FileAccess));
-            if (FileAccess == FileAccess.Write)
+            // TEMP: If the target type is set, use the binder logic to get a value of that type.
+            if (context.TargetType != null)
```
Comment (MikeStall) [inline src/WebJobs.Script/Binding/BlobBinding.cs]: Can we get rid of this check and always set TargetType? For non-C# languages, it's Stream (based on the 'else' block).

File: src/WebJobs.Script.WebHost/Controllers/HomeController.cs (hunk 1 above)
Comment (fabiocav) [inline src/WebJobs.Script.WebHost/Controllers/HomeController.cs]: I believe this was added by @mathewc , so he would have more context.

**Comment by fabiocav** [comment]
Closing this PR since #32 has been opened against 'master'.

**Pull request closed without merging**